### PR TITLE
Multiple Error and Enum Docstrings Alignment

### LINKED ISSUE #1074: Align docstring formatting across all functions with multiple errors of the same type and across all Enum classes
Numpy does not specify docstring writing instructions for functions that raise multiple errors of the same type nor for Enum classes.
Go through the code base and find all functions that raise multiple errors of the same type and match the formatting to the agreed-upon formatting. Do the same for all the Enum classes.

## Expected Behavior
In the docstrings for a function that raises multiple errors of the same type, the formatting should be as follows:
```
Raises
------
ValueError
    Reason 1
    Reason 2
```

For all Enum classes, the docstrings should look like this:
```
class YourEnumClass(Enum):
"""YourEnumClass description
Attributes
----------
Attribute1 : str
    Represents...fill in what Attribute1 represents here.
Attribute2 : str
    Represents...fill in what Attribute2 represents here.
```

## Current Behavior
Currently there are multiple variations on how we are writing docstrings for Enum classes and for functions that raise multiple errors of the same type.

## Possible Solution
Searching the code base for `(Enum)` to find the Enum classes.

Searching the code base for `Raises` to find the cases where functions raise errors and looking at each case to see if there are multiple errors of the same type being raised e.g.

```
Raises
------
ValueError
    Reason 1
ValueError
    Reason 2
ValueError
    Reason 3
```

should become:

```
Raises
------
ValueError
    Reason 1
    Reason 2
    Reason 3
```

### PR DESCRIPTION
**Pull request by ew3361zh**
PR to address issue #1074.

This PR attempts to align the docstring formatting for functions that raise multiple instances of the same type of error and additionally aligns enum-based class docstrings to use the "Attributes" docstring category as is outlined in the issue.

Notes:
- I made some small updates to syntax ad grammar where I saw it necessary (a period at the end of a sentence, making a sentence clearer, making sure the variables referenced in the error raised were correctly spelled, etc).
- There were some functions that raised errors that didn't list the errors raised in the docstrings and I added those in.
- There was one enum class in the Animal module (AnimalGRoupingScenario) that had a TODO for the docstrings so I didn't update the Attributes formatting.
- I know Matthew is also working on docstring formatting in the C&S module so I hopefully am not duplicating any work he's doing right now.

### CODE REVIEW

File: RUFAS/routines/field/crop/heat_units.py
Comment (matthew7838) [inline RUFAS/routines/field/crop/heat_units.py:1]: This file duplicates, but you can just accept what's on the main branch now to solve the merge conflicts and it should be fine.

File: RUFAS/routines/field/soil/snow.py [hunk 1]
```diff
@@ -161,6 +161,7 @@ def update_snow(self, current_day_conditions: CurrentDayConditions, day: int) ->
         Raises
         ------
         ValueError
+            If the snow_content is < 0.0.
```
Comment (matthew7838) [inline RUFAS/routines/field/soil/snow.py:164]: Need a blank line between end of docstring and closing `"""`.

File: RUFAS/routines/animal/life_cycle/cow.py [hunk 2]
```diff
@@ -622,21 +622,34 @@ def get_bw_change(self, CI):  # noqa
...
+            production and fat percent, for temporary use.
+        bool
+            cull_stage which is True if a cow is culled, False if it stays in the herd.
+        bool
+            new_born status which is True if a calf is born.
+
+        Raises
+        ------
+        ValueError
+            If reproduction program not in list of current reproduction programs.
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/cow.py:652]: Need a blank line between the end of docstring and closing `"""`.

File: RUFAS/routines/animal/animal_types.py [hunk 3]
```diff
@@ -2,7 +2,24 @@
...
+    HEIFER_II : str
+        A heifer that is either eligible for breeding (based on user-inputted Breeding Start Day for heifers),
+        or in early pregnancy.
+    HEIFER_III : str
+        A close-up heifer (a heifer within the user-defined close-up period, i.e. Prefresh Day).
+    DRY_COW : str
+        A cow in the stage of their lactation cycle where milk production ceases prior to calving.
+    LAC_COW : str
+        A lactating cow.
+    """
```
Comment (matthew7838) [inline RUFAS/routines/animal/animal_types.py]: Missing extra line.

File: RUFAS/routines/animal/animal_combinations.py [hunk 4]
```diff
@@ -5,18 +5,19 @@ class AnimalCombination(Enum):
...
+        Represents a pen with growing heiferI and heiferII only.
     CLOSE_UP: int
-        Represents a pen with heiferIII and dry cows only
+        Represents a pen with heiferIII and dry cows only.
     LAC_COW: int
-        Represents a pen with lactating cows only
+        Represents a pen with lactating cows only.
     GROWING_AND_CLOSE_UP: int
-        Represents a pen with all heifers and dry cows only
+        Represents a pen with all heifers and dry cows only.
```
Comment (matthew7838) [inline RUFAS/routines/animal/animal_combinations.py:20]: Missing extra line.

File: RUFAS/output_manager.py [hunk 5]
```diff
@@ -1059,8 +1067,10 @@ def load_variables_pool_from_file(self, file_path: Path) -> None:
 
         Raises
         ------
-        Exception
-            If an error occurs while opening or reading the user-provided file path.
+        FileNotFoundError
+            If the variables pool file does not exist at the specified path.
+        json.JSONDecodeError
+            If there is an error in decoding the JSON file.
```
Comment (matthew7838) [inline RUFAS/output_manager.py:1074]: Missing extra line.

File: RUFAS/output_manager.py [hunk 6]
```diff
@@ -496,7 +504,7 @@ def _list_to_file_txt(self, data_list: List[str], path: str) -> None:
         Raises
         ------
         Exception
-            If an error occurs while saving to the file
+            If an error occurs while saving to the file.
```
Comment (matthew7838) [inline RUFAS/output_manager.py:507]: Missing extra line.

File: RUFAS/input_manager.py [hunk 7]
```diff
@@ -194,6 +206,11 @@ def _populate_pool(self, eager_termination: bool) -> bool:
         -------
         bool
             True if data is valid, otherwise False.
+
+        Raises
+        ------
+        KeyError
+            If faulty data type found in data blob key.
```
Comment (matthew7838) [inline RUFAS/input_manager.py:215]: Missing extra line.

File: RUFAS/input_manager.py [hunk 8]
```diff
@@ -161,6 +166,13 @@ def _load_data_from_csv(self, file_path: str) -> Dict[str, Any]:
         -------
         Dict[str, Any]
             The data dictionary loaded from the json file.
+
+        Raises
+        ------
+        FileNotFoundError
+            If the CSV file does not exist at the specified path.
+        Exception
+            For any other unexpected errors during CSV file loading.
```
Comment (matthew7838) [inline RUFAS/input_manager.py:176]: Missing extra line.

File: RUFAS/input_manager.py [hunk 9]
```diff
@@ -135,6 +135,11 @@ def _load_data_from_json(self, file_path: str) -> Dict[str, Any]:
         -------
         Dict[str, Any]
             The data dictionary loaded from the json file.
+
+        Raises
+        ------
+        Exception
+            For any other unexpected errors during JSON file loading.
```
Comment (matthew7838) [inline RUFAS/input_manager.py:142]: Missing extra line.

File: RUFAS/routines/animal/life_cycle/heiferII.py [hunk 10]
```diff
@@ -1019,6 +1019,11 @@ def _set_up_hormone_schedule(self, animal_category: Literal['heifers', 'cows'],
         Returns
         -------
         None
+
+        Raises
+        ------
+        Exception
+            If there is no hormone delivery schedule for the animal category.
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/heiferII.py:1026]: Missing extra line.

File: RUFAS/routines/animal/life_cycle/heiferII.py [hunk 11]
```diff
@@ -1116,6 +1121,11 @@ def _set_repro_program(self, sim_day: int, repro_program: str) -> None:
         Returns
         -------
         None
+
+        Raises
+        ------
+        ValueError
+            If the reproduction program is not valid.
```
Comment (matthew7838) [inline RUFAS/routines/animal/life_cycle/heiferII.py:1129]: Missing extra line.

File: RUFAS/routines/animal/pen.py [hunk 12]
```diff
@@ -688,6 +688,10 @@ def _get_prefix_and_default_manure_excretion(animal: Calf | HeiferI | HeiferII |
         Tuple[str, AnimalManureExcretions]
             A tuple containing the prefix and default manure value for the animal.
 
+        Raises
+        ------
+        ValueError
+            If prefix for animal type is not found.
```
Comment (matthew7838) [inline RUFAS/routines/animal/pen.py:694]: Missing extra line.

File: RUFAS/routines/animal/purchased_feed_emissions_estimator.py [hunk 13]
```diff
@@ -81,6 +81,22 @@ def create_daily_purchased_feed_emissions_report(self, daily_feed_totals: dict[s
...
+
+        Raises
+        ------
+        ValueError
+            If the location is not found.
+        ValueError
+            If the return value from the FCC's API is null.
+        requests.exceptions.RequestException
+            If the max attempts to reach FCC's API was reached and all attempts failed.
+        """
```
Comment (matthew7838) [inline RUFAS/routines/animal/purchased_feed_emissions_estimator.py:100]: Missing extra line.

File: RUFAS/routines/animal/ration/ration_driver.py [hunk 14]
```diff
@@ -1006,6 +1006,11 @@ def get_feed_data_from_feed_ids(self, feed_ids: Set[int]):
         -------
         A dictionary that contains a subset of data from all the available feeds based on the
         given set of feed ids
+
+        Raises
+        ------
+        KeyError
+            If the feed id is not found in the feed id list.
```
Comment (matthew7838) [inline RUFAS/routines/animal/ration/ration_driver.py:1013]: Missing extra line.

File: RUFAS/routines/feed_storage/harvested_crop.py [hunk 15]
```diff
@@ -66,6 +66,11 @@ class HarvestedCrop:
     def __post_init__(self):
         """
         Validates that the type of the crop is consistent with its category.
+
+        Raises
+        ------
+        ValueError
+            If the crop type is not valid for the crop category.
```
Comment (matthew7838) [inline RUFAS/routines/feed_storage/harvested_crop.py:73]: Missing extra line.

File: RUFAS/routines/manure/manure_separators/manure_separator_classes.py [hunk 16]
```diff
@@ -15,7 +15,32 @@
...
+        The slope screen method of manure separation.
+    MECHANICAL_SAND_SEPARATOR : str
+        The mechanical sand separator method of sand-manure separation.
+    SAND_LANE_MANURE_SEPARATION : str
+        The sand lane method of sand-manure separation.
+
+    Notes
+    -----
+    The default organic separator is ROTARY_SCREEN.
+    The default sand lane separation method is SAND_LANE_MANURE_SEPARATION.
```
Comment (matthew7838) [inline RUFAS/routines/manure/manure_separators/manure_separator_classes.py:42]: Missing extra line.

File: RUFAS/routines/manure/manure_treatments/manure_treatment_types.py [hunk 17]
```diff
@@ -2,7 +2,35 @@
...
+    COMPOST_BEDDED_PACK_BARN : str
+        Compost bedded pack barn.
+    OPEN_LOTS : str
+        Open lots.
+    SEPARATED_SOLIDS_STORAGE : str
+        Separated solids storage.
+
+    Notes
+    -----
+    The DEFAULT manure treatmenet type is SLURRY_STORAGE_UNDERFLOOR
```
Comment (matthew7838) [inline RUFAS/routines/manure/manure_treatments/manure_treatment_types.py:32]: Missing extra line.

File: RUFAS/routines/manure/manure_treatments/manure_types.py [hunk 18]
```diff
@@ -5,13 +5,12 @@ class ManureType(Enum):
...
     This is an Enum class that represents different types of manure.
 
-    Attribute
+    Attributes
     ----------
     LIQUID : str
         Represents liquid manure.
     SOLID : str
         Represents manure in solid form.
-
```
Comment (matthew7838) [inline RUFAS/routines/manure/manure_treatments/manure_types.py:14]: Missing extra line.

File: RUFAS/util.py [hunk 19]
```diff
@@ -86,8 +86,8 @@ def read_json_file(file_path: Path):
 
         Raises
         ------
-        InvalidJSONFileError: If the json file at the given path does not
-                conform with the format required
+        InvalidJSONFileError:
```
Comment (matthew7838) [inline RUFAS/util.py]:
```suggestion
        InvalidJSONFileError
```

**Review by matthew7838 — CHANGES_REQUESTED** [review]
Great improvements Niko! There are some parts where the extra line between the content of docstrings and closing `"""` is missing. I have pointed those out. Should be minor, once those are fixed should be good to go.

**Review by matthew7838 — APPROVED** [review]
LGTM, great work Niko. Thanks for the modification.

**Review by allisterakun — APPROVED** [review]
Good work, Niko! Nicely done~

**Pull request merged**